server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsRequest.java

```diff
@@ -44,6 +46,9 @@ public NodesStatsRequest(StreamInput in) throws IOException {
         indices = new CommonStatsFlags(in);
         requestedMetrics.clear();
         requestedMetrics.addAll(in.readStringCollectionAsList());
+        if (in.getTransportVersion().onOrAfter(TransportVersions.NEED_SHARDS_STATS_ADDED)) {
+            needShardsStats = in.readBoolean();
+        }
```

Comment:
Could we set it to `true` in an `else` branch here? I know it defaults to `true` in the field declaration, but I think readers might miss that and assume that older versions default to `false` since there's no `else` branch here.

Reply:
Sure, this is more readable. I added it in c31ecc73120ca65516e27195a5969efae1d9b898